File: Backend/codewithkatz/src/main/java/com/katziio/codewithkatz/entity/editor/Editor.java
package com.katziio.codewithkatz.entity.editor;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Editor {
    private Long id;
    private String name;
    private String email;
    private String password;
    private int age;
    private String country;
    private String gender;
    private int experienceInYears;
    private String profilePictureUrl;
    private String phone;
    private String portfolioLink;
    private Date profileCreatedAt;
    private Date ProfileUpdatedAt;
    private List<EditorCertification> editorCertifications;
    private List<EditorCommunicationLanguage> editorCommunicationLanguages;
    private List<EditorPreferences> editorPreferences;
    private List<EditorProject> editorProjects;
    private List<EditorSampleVideo> editorSampleVideos;
    private List<EditorSocialMedia> editorSocialMedia;
}

File: Frontend/CollabWithKatz/src/app/common/editor/editor.ts
export class Editor {
        id!: number;
        name!: string;
        email!: string;
        password!: string;
        age!: number;
        country!: string;
        gender!: string;
        experienceInYears!: number;
        profilePictureUrl!: string;
        phone!: string;
        portfolioLink!: string;
        profileCreatedAt!: Date;
        ProfileUpdatedAt!: Date;
        editorCertifications: EditorCertification[] = [];
        editorCommunicationLanguages: EditorCommunicationLanguage[] = [];
        editorPreferences: EditorPreferences[] = [];
        editorProjects: EditorProject[] = [];
        editorSampleVideos: EditorSampleVideo[] = [];
        editorSocialMedia: EditorSocialMedia[] = [];
        editorSkills: EditorSkill[] = [];
}

export class EditorCertification {
        id!:number;
        certificationName!:string;
        certificationLink!:string;
}

export class EditorCommunicationLanguage {
        id!: number;
        language!: string;
        languageProficiency!: number;
}

export class EditorPreferences {
        id!: number;
        projectPreferences!: string;
}

export class EditorProject {
        id!: number;
        projectName!: string;
        projectStatus!: boolean;
}

export class EditorSampleVideo {
        id!: number;
        videoName!: string;
        videoLink!: string;
}

export class EditorSocialMedia {
        id!: number;
        platform!: string;
        platformLink!: string;
}

export class EditorSkill {
        id!:number;
        aboutSkill!:string;
        availability!:string;
        flexibility!:string;
        technicalProficiency!:string;
        equipmentDetails!:string;
}
